src/Pairs.cpp

```diff
@@ -329,6 +337,7 @@ bool CPairs::AddPair(bool enabled, const std::wstring& orig, const std::wstring&
     pd.m_fat          = fat;
     pd.m_compressSize = compressSize;
     pd.m_syncDeleted  = syncDeleted;
+    pd.m_ResetOriginalArchAttr = ResetOriginalArchAttr;
 
     // make sure the paths are not root names but if root then root paths (i.e., ends with a backslash)
```

Comment:
I assume this feature is also needed when encrypting files, not just when copying them.
See the EncryptFile method.
Also, check the "set the file timestamp" comment

Reply:
Thanks, I added code to implement the feature when encrypting files upon return from EncryptFile method (lines 492, 648, 821) because pt.m_ResetOriginalArchAttr is unknow with the EncryptFile method. If you prefer I can add it as an argument to the method and put the code there once. Let me know and I will do as you prefer.

I am working on the file timestamp comment and others you submitted.

Reply:
I would pass it as an argument to the EncryptFile method.